src/main/java/org/mskcc/cmo/metadb/service/impl/ValidRequestCheckerImpl.java

```diff
@@ -0,0 +1,271 @@
+                requestJsonMap.replace("samples", newSamplesString);
+                LOG.info("CMO request passed sanity checking - missing samples...");
+                requestStatusLogger.logRequestStatus(requestJson,
+                        RequestStatusLogger.StatusType.CMO_REQUEST_MISSING_REQ_FIELDS);
+            }
+            return mapper.writeValueAsString(requestJsonMap);
+        }
+        LOG.error("Request failed sanity checking - logging request status...");
+        requestStatusLogger.logRequestStatus(requestJson,
+                RequestStatusLogger.StatusType.CMO_REQUEST_FAILED_SANITY_CHECK);
+        return null;
+    }
+
+    /**
+     * isValidCmoSample receives a sampleMap
+     * 
+     * <p><ul>
+     * <li>Checks if the sample has all the required fields
+     * <li>If RequestId, InvestigatorSampleId, CmoPatientId,
+     * SpecimenType, SampleType or NormalizedPatientId are missing, returns false
+     * </ul>
+     * @throws JsonProcessingException or JsonMappingException 
+     */
+    private boolean isValidCmoSample(Map<String, String> sampleMap,
+            boolean isCmoRequest, boolean hasRequestId) throws JsonMappingException, JsonProcessingException {
+        if (sampleMap.isEmpty() || sampleMap == null) {
+            return Boolean.FALSE;
+        }
+        if (isCmoRequest && !hasBaitSet(sampleMap)) {
+            return Boolean.FALSE;
+        }
+        if (!hasRequestId || !hasInvestigatorSampleId(sampleMap)
+                || !hasCmoPatientId(sampleMap) || !hasSpecimenType(sampleMap)
+                || !hasSampleType(sampleMap) || !hasNormalizedPatientId(sampleMap)) {
+            return Boolean.FALSE;
+        }
+        return Boolean.TRUE;
+    }
+
+    /**
+     * isValidNonCmoSample receives a sampleMap
+     * 
+     * <p><ul>
+     * <li>Check if the sample has all the required fields
+     * <li>If baitSet or normalizedPatientId is missing, returns false
+     * </ul>
+     * @throws JsonProcessingException or JsonMappingException 
+     */
+    private boolean isValidNonCmoSample(Map<String, String> sampleMap)
+            throws JsonMappingException, JsonProcessingException {
+        if (sampleMap.isEmpty()
+                || sampleMap == null
+                || !hasBaitSet(sampleMap)
+                || !hasNormalizedPatientId(sampleMap)) {
+            return Boolean.FALSE;
+        }
+        return Boolean.TRUE;
+    }
+
+    private boolean isCmoRequest(Map<String, Object> requestJsonMap) {
+        if (Strings.isBlank(requestJsonMap.get("cmoRequest").toString())) {
+            return Boolean.FALSE;
+        }
+        return Boolean.valueOf(requestJsonMap.get("cmoRequest").toString());
+    }
+
+    private boolean hasRequestId(Map<String, Object> requestJsonMap) {
+        return Strings.isBlank(requestJsonMap.get("requestId").toString());
+    }
+
+    private boolean hasBaitSet(Map<String, String> sampleMap) {
+        return Strings.isBlank(sampleMap.get("baitSet"));
```

Comment:
another note: 

if the sample map contains "baitSet" and it is non-empty then `Strings.isBlank(sampleMap.get("baitSet"))` will be returning false which i think is the opposite of what we want to return right?

Just add a `!` to the front of `Strings.`